src/Core/Models/Mail/MailQueueMessage.cs

```diff
@@ -9,7 +11,21 @@ public class MailQueueMessage : IMailQueueMessage
         public IEnumerable<string> BccEmails { get; set; }
         public string Category { get; set; }
         public string TemplateName { get; set; }
-        public object Model { get; set; }
+        private object _model;
+        public object Model
+        {
+            get
+            {
+                if (_model?.GetType() == typeof(JObject))
+                {
+                    return (_model as JObject).ToObject(ModelType);
+                }
+                return _model;
+            }
+
+            set => _model = value;
+        }
+        public Type ModelType { get; set; }
```

Comment:
Curious why we're not just using a dictionary instead of a JObject for the model body; OR, use JSON deserialization/Serialization to adapt to a Dictionary in both directions, then there shouldn't be a reason to use type switching...

Either way, I didn't think `Type` was JSON serialization, or if it is, would include waaaay too much info in the payload vs. using the `model.GetType().Name` or `model.GetType().FullName` instead, and fetching the type definition from the deserialization side.

Reply:
it did work, but this new way is way better -- thanks for the advice!